MAUI/Autocomplete/Searching-Filtering.md

```diff
@@ -171,6 +171,276 @@ autocomplete.TextSearchMode = AutocompleteTextSearchMode.Contains;
+        Cities.Add(new CityInfo() { CityName = "Chennai", CountryName = "India", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "Andhra Pradesh", CountryName = "India", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "Kerala", CountryName = "India", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "Chicago", CountryName = "USA", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "Los Angels", CountryName = "USA", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "Houston", CountryName = "USA", IsCapital = false });
+        Cities.Add(new CityInfo() { CityName = "New York", CountryName = "USA", IsCapital = true });
+        Cities.Add(new CityInfo() { CityName = "Wasington", CountryName = "USA", IsCapital = false });
+        
+    }
+}
+{% endhighlight %}
+
+{% endtabs %}
+
+Now, let us create a custom filtering class to apply your filter logic to `Autocomplete` control by following these steps:
+
+**Step 1:** Create a class that derives from the `IAutocompleteFilterBehavior` interface.
+
+{% tabs %}
+{% highlight C# %}
+
+/// <summary>
+/// Represents a custom filtering behavior for `Autocomplete` control. 
+/// </summary>
+public class CityFilteringBehavior : IAutocompleteFilterBehavior
+{
+
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+**Step 2:** Then, implement the `GetMatchingIndexes` method of `IAutocompleteFilterBehavior` interface to create your own suggestion list (containing the indices of the filtered items) based on the text entered in the `Autocomplete` control that needs to be shown in the drop-down. The `GetMatchingIndexes` method contains following arguments:
+
+* `source` - The owner of the filter behavior, which holds information about `ItemsSource`  property and so on.
+* `filterInfo` - Contains details about the text entered in `Autocomplete` control. Using this text, you can prepare suggestion list, which gets displayed in the drop-down list. 
+
+The following image shows how to display cities in a drop-down based on the country name entered in the `Autocomplete` control.
+
+{% tabs %}
+{% highlight C# %}
+
+public class CityFilteringBehavior : IAutocompleteFilterBehavior
+{
+    public async Task<object> GetMatchingItemsAsync(SfAutocomplete source, AutocompleteFilterInfo filterInfo)
+    {
+         IEnumerable itemssource = source.ItemsSource as IEnumerable;
+        var filteredItems = (from CityInfo item in itemssource
+                             where item.CountryName.StartsWith(filterInfo.Text, StringComparison.CurrentCultureIgnoreCase) ||
+                                   item.CityName.StartsWith(filterInfo.Text, StringComparison.CurrentCultureIgnoreCase)
+                             select item);
+
+        return await Task.FromResult(filteredItems);
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+**Step 3:** Applying custom filtering to the `Autocomplete` control by using the `FilterBehavior` property.
```

Comment:
**Step 3:** Apply the custom filtering to the `Autocomplete` control by using the `FilterBehavior` property.